GoogleForCreators/web-stories-wp
# Dashboard: Now with Poster Images and Faster Content Rendering

## Context

Part of performance sprint. 

## Summary

The Dashboard was loading the entire story for each of a user's stories. THIS IS SO MUCH TO RENDER. 
Swapping it out for poster images. This makes the presentation consistent with the block view in posts and helps allows us to avoid rendering a gluttony of nodes. 

Downside is on hover you don't get to "preview" any animation or see the story, feel like this is a fine trade off. If we feel like that's being missed by users we can either add a story detail view (like the templates) or find a way to draw attention to the preview button in the context menus. 

## Further ideas for improvement
- Possibly use `srcset` for poster images 
- There's gotta be a way to kill the `usePageView` that's running on resize to calculate the grid size if we can just make the images use `aspect-ratio` or something, i'm not sure this is a massive win. 
- I'm curious if there's more we can do for first paint, as this focuses on additional renders in the dashboard. 

## Relevant Technical Choices

- Cut the rendering time of StoryGridView from 8.6ms (Main) to 2.7ms (or 1.4ms on various runs) (This Branch) when throttled to use Slow 3G network connection. 
- Cut out an entire extra fetch of stories when toggling between Grid and List views, there's no need to go get that data again. That saves us about 0.8ms render time in slow 3G network connections 

| Updated & Slow 3G | Main No Throttle | Main 3G Slow | 
| --- | --- | --- |
| <img width="1212" alt="Screen Shot 2021-07-13 at 11 55 51 AM" src="https://user-images.githubusercontent.com/10720454/125509379-2e0471ad-f6b4-424b-8b30-080ad47f0eb8.png"> | <img width="1239" alt="Screen Shot 2021-07-13 at 10 35 15 AM" src="https://user-images.githubusercontent.com/10720454/125508614-e39f7207-8d9b-4cda-96c2-bd8a5c7bda4c.png"> | <img width="1248" alt="Screen Shot 2021-07-13 at 10 43 51 AM" src="https://user-images.githubusercontent.com/10720454/125508565-79a89e3f-71c4-4987-94ed-73374147869f.png"> |


- I pulled apart the Story Grid components a little bit so that they could be smaller and we could memoize some of the data, like the context menus. This doesn't do a lot, but i think it makes sense and really doesn't hurt. The directory structure here was out of date anyways since these grids aren' "shared". 
- Since this grid is no longer shared with templates (which still have to use the amp structure and all the providers) I gutted the unnecessary parts of the layout so we're not rendering excess DOM nodes for each grid item. 
- Added a new height to the `usePageView` hook so that we can get poster images rendered as `3:4`. 

## To-do

- [x] Make sure I didn't kill the templates views 
- [x] Clean up the list view with same changes 
- [x] Fix what I broke on the grid focus trap
- [x] Organize analytics of perf improvements 
- [x] Organize further improvements 
- [x] Updates to tests

## User-facing changes

- Dashboard loads story posters instead of story previews - new ratio of 3:4 
- A11y update to grid navigation, which I think is more straight forward. There's nothing on the story poster to interact with that isn't already in the context menu. Now, navigating the grid, because there's no 'active preview mode' to deal with we can just use a list and arrow through the context menus and get all the info we need. 
<img width="1229" alt="Screen Shot 2021-07-13 at 11 49 56 AM" src="https://user-images.githubusercontent.com/10720454/125508484-bf2a39b6-3eef-4a69-806f-32c1239732e9.png">


## Testing Instructions

Verify that the dashboard still works as expected with a mouse
Verify the above a11y keyboard nav changes - should be way less complicated now. 
Make sure that the templates view and list views work. 

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #5147
